pkg/operator/ceph/file/filesystem.go

```diff
@@ -116,3 +124,156 @@ func validateFilesystem(context *clusterd.Context, f cephv1.CephFilesystem) erro
 
 	return nil
 }
+
+// Filesystem represents an instance of a Ceph file system (CephFS)
+type Filesystem struct {
+	Name           string
+	metadataPool   *model.Pool
+	dataPools      []*model.Pool
+	activeMDSCount int32
+}
+
+// newFS creates a new instance of the file (MDS) service
+func newFS(name string, metadataPool *model.Pool, dataPools []*model.Pool, activeMDSCount int32) *Filesystem {
+
+	metadataPool.Name = fmt.Sprintf("%s-%s", name, metadataPoolSuffix)
+	for i, pool := range dataPools {
+		pool.Name = fmt.Sprintf("%s-%s%d", name, dataPoolSuffix, i)
+	}
+
+	return &Filesystem{
+		Name:           name,
+		metadataPool:   metadataPool,
+		dataPools:      dataPools,
+		activeMDSCount: activeMDSCount,
+	}
+}
+
+// doFilesystemCreate starts the Ceph file daemons and creates the filesystem in Ceph.
+func (f *Filesystem) doFilesystemCreate(context *clusterd.Context, clusterName string) error {
```

Comment:
This method is quite big and has several built-in tests in it. Perhaps you could try to break it into smaller pieces so we can test it more easily too? I haven't found any test for it.

Reply:
It's moved from the `daemon` code path and renamed from `CreateFilesystem`. The `object` and `file` dirs do need some cleanup, but I'd rather do that separately.

Reply:
I see, so leave it as is then.